### ISSUE TITLE
NuGet restore: MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated

### ISSUE THREAD

**Issue by koszeggy** [filipnavara/mac-playground#34]
I performed the following steps so far, in order to attempt to build mac-playground:
* I installed a fresh macOS Big Sur on VMWare
* I installed the latest Visual Studio for mac,
* I could not open the solution until I installed the latest .NET 6 preview SDK
* Build failed saying there is no Xcode so I created an Apple ID, downloaded and installed Xcode, too
* The Xcode path then was wrong, finally I managed to find out how to fix it in Preferences

Now I can build the `System.Private.CoreLib` project but nothing else. The build fails with `NuGet packages need to be restored...` but it is unable to restore the packages:

```
/usr/local/share/dotnet/sdk-manifests/6.0.100/microsoft.net.sdk.macos/WorkloadManifest.targets(7,3) : error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. Invalid path
Unable to create package spec for project. '/Users/username/repo/mac-playground/FormsTest/FormsTest.csproj'
```

And I'm stuck here. How could I overcome this issue? Are there any further gotchas I need to know about?

Btw, I wish the short step-by-step guide in the README was a bit more detailed (feel free to use my points above). The second bullet point there says I need to choose "Xamarin - Debug" but I only have the regular "Debug"/"Release" options so I'm not even sure whether I reached the second step in the build process...

**Comment by filipnavara** [comment]
Please refer to the issue #33 and try the [build-fixes branch](https://github.com/emclient/mac-playground/tree/build-fixes). I still hope to get this into a shape where it would build out of the box but the idea is to do that along with multi-targeting .NET 6.

**Comment by koszeggy** [comment]
Thank you, using the fix-build branch and the issue #33 helped to step further.

I would then just add the last steps as a summary:
* ...
* Switch to the `build-fixes` branch
* Open `Mono.WebBrowser/Mono.WebBrowser.csproj`, replace `netstandard2.1` with `netstandard2.0`
* From the repo root: `nuget restore`, then `msbuild`
* Now you can launch `FormsTest/bin/Debug/xamarinmac20/osx-x64/FormsTest.app/Contents/MacOS/FormsTest`

Finding out the last step took me my last 2 hours because from Visual Studio I always get an exception:
```
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.FileNotFoundException: Could not load file or assembly 'System, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e' or one of its dependencies.
File name: 'System, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'
```

There is no stack trace, as the app can't even start. I tried to rebuild everything, I removed the complete repo and cloned it again, I removed even .NET 6 so now I have only the latest stable .NET 5. Is this an expected issue or is it possible to run or debug the project from Visual Studio?

(Anyhow, thank you for your efforts making all this possible)

**Comment by filipnavara** [comment]
I'll try to restore the build from VS once .NET 6 Preview 6 comes out. We have it working locally inside a bigger VS solution and it's not trivial to isolate what breaks the build from inside VS.

**Comment by filipnavara** [comment]
I re-synced the repository with our upstream version. The `build-fixes` branch now builds for me from VS4Mac 8.10.1 (build 71). I just open the solution, switch project to "FormsTest.app", keep configuration as `Debug` and hit Run.

**Comment by filipnavara** [comment]
This should be fixed by now.
